node-data/src/ledger.rs

```diff
@@ -99,6 +99,9 @@ pub struct SpentTransaction {
 
 impl Transaction {
     pub fn hash(&self) -> [u8; 32] {
```

Comment:
Shouldn't we remove/make private `hash(..)` method to avoid its misuse?

Reply:
Hash is currently used by the consensus to create the merkletree, so I cannot remove or reduce the visibility.
I can move this outside or just add some doc comments explaining the differences

Reply:
> Consequently, if a block were to be manipulated to include a transaction with an invalid proof, it would pass the initial check but fail during acceptance (so no harm for the network)

My (maybe mis)understanding here is that we also need to use `tx_id()` for calculating `merkletree` in the (consensus) Block generator too.